src/reference.rs

```diff
@@ -65,6 +65,20 @@ impl<'repo> Reference<'repo> {
         unsafe { raw::git_reference_is_valid_name(refname.as_ptr()) == 1 }
     }
 
+    /// Ensure the reference name is well-formed.
+    pub fn name_is_valid(refname: &str) -> Result<bool, Error> {
+        crate::init();
+        let refname = CString::new(refname)?;
+        let mut valid: libc::c_int = 0;
+        unsafe {
+            try_call!(raw::git_reference_name_is_valid(
+                &mut valid,
+                refname.as_ptr()
+            ));
+        }
+        Ok(valid == 1)
```

Comment:
Since this is identical to `git_reference_is_valid_name`, instead of adding a new method, can you just update the `is_valid_name` to call the new `git_reference_name_is_valid` function?

Otherwise, it seems quite confusing to have two nearly identically named functions which do the exact same thing.

Reply:
@ehuss You bring up a good point, though one concern I have here is deciding between:
* Silently gobbling up error and panicking as done in the existing implementing by calling unwrap (`let refname = CString::new(refname).unwrap();`)
* Updating the existing contract (and make a breaking change) by returning a Result as opposed to a scalar bool

WDYT?

Reply:
Hm. I believe the only way `git_reference_name_is_valid` can return an error is on OOM, right? I would be inclined to just panic if `git_reference_name_is_valid` fails. I probably wouldn't bother with changing the existing behavior with `CString::new`, but if I were starting from scratch, I would also change that to return `false` since internal nuls aren't really valid.

I think perhaps in a future semver breaking release, the method could be changed to return a `Result`, but I probably wouldn't bother.

Reply:
@ehuss I've made the changes -- I've also added comments to the lib.rs file to ensure there is no confusion around why we don't use a given binding.

Reply:
@ehuss Thoughts?

Reply:
I'm a bit confused by the current PR since it seems to still have two methods doing the same thing.  Would it be possible to keep just `is_valid_name` with its signature and panic if `git_reference_name_is_valid` returns an error? I'm not sure the error return is particularly useful here, and I'd like to avoid complicating things.